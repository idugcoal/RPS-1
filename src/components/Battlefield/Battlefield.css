.container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.bigContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 50px;
}
