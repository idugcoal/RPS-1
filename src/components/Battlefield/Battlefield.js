import React from 'react';
import Weapon from '../Weapon/Weapon';
import { computeWinner } from '../../utils';
import { numToWeaponMap, numToResultMap, numToDetailResultMap } from '../../constants';
import style from './Battlefield.css';

// This is the default battlefield for player vs computer
const Battlefield = ({
  firstName,
  firstSelection,
  secondSelection,
}) => {
  let winner = '';
  let result = '';
  let detailedResult = '';
  let firstWeapon = 'questionmark';
  let secondWeapon = 'questionmark';
  if (firstName !== undefined &&
      firstSelection !== undefined &&
      secondSelection !== undefined) {
    winner = computeWinner(firstSelection, secondSelection);
    result = `${firstName} ${numToResultMap[winner]}`;
    detailedResult = firstSelection !== undefined ?
      numToDetailResultMap[firstSelection][secondSelection] : null;
    firstWeapon = `${numToWeaponMap[firstSelection]}`;
    secondWeapon = `${numToWeaponMap[secondSelection]}`;
  }
  return (
    <div className={style.outerContainer} >
      <div className={style.resultDiv}>
        {result}
      </div>
      <div className={style.innerContainer}>
        <Weapon name={firstWeapon} />
        VS
        <Weapon name={secondWeapon} />
      </div>
      <div className={style.resultDiv}>
        {detailedResult}
      </div>
    </div>
  );
};

export default Battlefield;
