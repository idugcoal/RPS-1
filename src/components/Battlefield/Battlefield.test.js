import React from 'react';
import { shallow } from 'enzyme';
import Battlefield from './Battlefield';

test('should render correctly with no props', () => {
  const wrapper = shallow(<Battlefield />);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly with name and selection props', () => {
  const wrapper = shallow(<Battlefield
    firstName='You'
    firstSelection={0}
    secondSelection={1}
  />);
  expect(wrapper).toMatchSnapshot();
});
