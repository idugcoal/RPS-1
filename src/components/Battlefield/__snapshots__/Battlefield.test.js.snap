// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with name and selection props 1`] = `
<div>
  <div>
    You Lose
  </div>
  <div>
    <Weapon
      name="rock"
    />
    VS
    <Weapon
      name="paper"
    />
  </div>
  <div>
    Paper covers Rock
  </div>
</div>
`;

exports[`should render correctly with no props 1`] = `
<div>
  <div />
  <div>
    <Weapon
      name="questionmark"
    />
    VS
    <Weapon
      name="questionmark"
    />
  </div>
  <div />
</div>
`;
